scripts/tests/avh/helpers/avh_instance.py

```diff
@@ -0,0 +1,240 @@
+                raise Exception(
+                    f"Timed out waiting for OS to boot for instance id {self.instance_id}",
+                    f"Did not find {booted_output} in {console_log}",
+                )
+
+            time.sleep(1.0)
+
+    def ssh_client(self, timeout=30):
+        if self.ssh_pkey is None:
+            self.ssh_pkey = paramiko.ecdsakey.ECDSAKey.generate()
+
+            self.ssh_key_id = self.avh_client.create_ssh_project_key(
+                self.name, f"{self.ssh_pkey.get_name()} {self.ssh_pkey.get_base64()}"
+            )
+
+        instance_quick_connect_command = self.avh_client.instance_quick_connect_command(
+            self.instance_id
+        )
+
+        split_instance_quick_connect_command = instance_quick_connect_command.split()
+        proxy_username = split_instance_quick_connect_command[-2].split("@")[-2]
+        proxy_hostname = split_instance_quick_connect_command[-2].split("@")[-1]
+        instance_ip = split_instance_quick_connect_command[-1].split("@")[-1]
+
+        ssh_proxy_client = paramiko.SSHClient()
+        ssh_proxy_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        ssh_client = paramiko.SSHClient()
+        ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        ssh_proxy_client.connect(
+            hostname=proxy_hostname,
+            username=proxy_username,
+            pkey=self.ssh_pkey,
+            look_for_keys=False,
+            timeout=timeout,
+        )
+
+        try:
+            proxy_sock = ssh_proxy_client.get_transport().open_channel(
+                kind="direct-tcpip",
+                dest_addr=(instance_ip, 22),
+                src_addr=("", 0),
+                timeout=timeout,
+            )
+
+            ssh_client.connect(
+                hostname=instance_ip,
+                username=self.username,
+                password=self.password,
+                sock=proxy_sock,
+                timeout=timeout,
+                look_for_keys=False,
+            )
+        except Exception:
+            raise Exception(
+                f"Failled to connect to {instance_ip} via SSH proxy {proxy_username}@{proxy_hostname}"
+            )
+
+        return ssh_client
+
+    def delete(self):
+        if self.console is not None:
+            self.console.close()
+
+            self.console = None
+
+        if self.ssh_key_id is not None:
+            self.avh_client.delete_ssh_project_key(self.ssh_key_id)
+
+            self.ssh_key_id = None
+
+        if self.instance_id is not None:
+            self.avh_client.delete_instance(self.instance_id)
+
+    def wait_for_state_deleted(self, timeout=60):
+        if self.instance_id is None:
+            return
+
+        start_time = time.monotonic()
+
+        while True:
+            try:
+                self.avh_client.instance_state(self.instance_id)
+            except avh_api.exceptions.NotFoundException:
+                break
```

Comment:
log or add a comment explaining why this is ok to ignore

Reply:
Added comment